### ISSUE TITLE
checking off dailies instead unchecks other dailies

### ISSUE THREAD

**Issue by Suzaqu** [HabitRPG/habitica#6670]
When I check off an uncompleted task, the completed task above it (or sometimes 2-3 tasks above it) will become unchecked again, and gold and experience removed. This has happened several times in the past few weeks, especially at the end of the day when I have only a few dailies left to complete. It seems to be some sort of issue with the cursor not lining up with the task correctly. Upon refreshing, the problem doesn't always resolve itself, tonight I had to refresh 5 or 6 times to get it to stop so I could recheck the tasks.

_edit by admin:_
### What to do if this happens to you:

**This problem is caused by a temporary corruption in the data that your browser has stored locally. To fix it, log out of Habitica (that forces your browser to remove the stored data).**

**If for any reason you can't get to the log out menu option, go to https://habitica.com/static/clear-browser-data and click the button there.**

**Comment by Alys** [comment]
Sorry about this. :( That must be really frustrating. The next time it happens, could you please try logging out of the Habitica website? If it still happens after logging back in, try logging out again and this time clearing your browser's cookies before logging back in.

Does that make the problem go away immediately or for longer?

If you aren't sure what your login details are, before logging out you can find them from Settings > Site (Settings is the gear-shaped icon on the top right of the website).

**Comment by KatAsh** [comment]
I am having the exact same issue and it's driving me crazy. I have tried doing multiple refreshes, logging out, logging in, sometimes it goes away and sometimes it keeps doing it. It's dropped me a level and removed multiple gold/mana/experience at various points over the last couple of weeks because of this, not to mention costing a huge amount of time and frustration. Today I've just given up and signed up to rest at the inn because I'm tired of fighting it. Is there a solution yet? Or any idea of what I might be able to do to get it to behave? I love this program.

**Comment by Alys** [comment]
KatAsh, so you're experiencing exactly the same thing that's described in the top comment here? When you mark one task as complete, another one above it immediately becomes marked incomplete - is that what happens?

**Comment by KatAsh** [comment]
Yes, if I click the first two tasks it's fine, after that it starts unclicking a task above every time I click the next ones. Also sometimes it won't 'grey out' a task when I click, or have a tick, but if I try reticking it it takes away gold/mana etc. Oddly now I've put myself as resting at the inn, it's checking the tasks off normally. Whether that's a fluke or not I don't know. And there was one day where I ticked everything off only to come back the next day and find everything had been unticked, I'd dropped a level, lost _a lot_ of gold and dropped my mana/lost costumes. That was annoying! Still love the program though :-)

**Comment by KatAsh** [comment]
Sorry I should have made it clear that it won't grey out or have a tick, ie, it'll look like an undone task, but if I click on it again it'll take away gold/mana etc

**Comment by Alys** [comment]
Sorry @KatAsh, that's sounds so disruptive. :(

You might be seeing two different problems.

_"there was one day where I ticked everything off only to come back the next day and find everything had been unticked"_ -- That sounds like this bug: https://github.com/HabitRPG/habitrpg/issues/4313 -- see the "Advice" section near the top for information about avoiding it and recovering from the damage. What happens is that sometimes your browser records your actions but doesn't send them to the server for permanent storage, so when cron runs (your start-of-day processing) it doesn't know that you completed your Dailies and it unfairly punishes you for them.

The weird clicking/unclicking behaviour might be related to that (your browser not properly syncing with the server so it doesn't cause completed tasks to go grey) or it could be something else, but it's probably browser-specific.

Can you please try using Habitica on a different browser and tell us if that makes a difference? If it's not possible to get a second browser on your PC, try using incognito mode in your normal browser, although I'm not certain that will work as well.

Which browser do you normally use for Habitica? Do you use Habitica on two different PCs (e.g., work and home maybe), and if so, do you see the same problem on both? Do you have the browser set up to sync its settings (bookmarks, history, etc) between the two PCs?

What's your User ID? See step 6 of Help > Report a Bug to find out how to find it.

**Comment by mornir** [comment]
"When I mark one task as complete, another one above it immediately becomes marked incomplete."
OR
"When I mark task A as complete, another one above it immediately becomes marked complete instead of task A."

For me, it happens so often that I stopped checking off dailies in my web browser and now use the iOS app instead (which works fine).
I use Habitica only on one PC with Chrome. With Firefox, I have the same issue.

In incognito mode, everything works fine!

**Comment by Alys** [comment]
@mornir When you notice that happening, please try logging out of Habitica and logging back in. It's likely that the problem is caused by corrupted local storage in your browser and logging out will clear local storage. 

Using incognito mode works because the website uses a different space for local storage than without incognito.

**Comment by KatAsh** [comment]
Hi Alys, sorry I didn't reply before, but there was no way to get the issue to behave consistently enough that I could tell if it was browser specific. But logging out and in or refreshing made zero difference as I said in my first post. What does, as I discovered myself from trying to stop loss of level etc, is going into Rest at the Inn mode. There's never a problem ticking things off when I'm there, and I've experimented further. If the page is refusing to cooperate going into Rest and then immediately out again is enough, I don't have to stay there to stop the glitch. I don't log out to 'reset' it, just press the Rest at Inn button and then press Leave Inn button and that's enough, I can continue normally. So is this likely to be a browser, local storage or other issue?

**Comment by Alys** [comment]
@KatAsh My recent logout advice was for mornir. It's likely to fix the problem for them since it does for most players who experience this. Can you see my post to you dated 27 days ago? I had some specific questions for you.

**Comment by KatAsh** [comment]
Hi Alys, I have it set up on one laptop only, a MacBook, and the browser is Safari. I could change browser but as I said it's so random that it might be days or weeks before I'd see the problem again -and that would mean me changing from a browser that I'm very happy with and that has never caused me any other problems for a problem I've now found a fix to. 
Again, going incognito wouldn't necessarily show any difference for days or weeks. It's totally random and it's only when it happens that it stayed happening, if that makes sense. 
But as I said yesterday if I click the Rest At The Inn button it fixes it immediately, and although it's not ideal it's quicker and easier for me than messing around with different browsers. 
I'm happy doing the fix, I only came back to mention it because I got the emails about mornir's problem and I thought it might be useful info for you to have for future reference. And because I'm curious as to why it works I guess.
